src/Scene.cpp

```diff
@@ -82,5 +86,109 @@ void Scene::run_frame()
     for (auto const& component : components_copy)
     {
         component->update();
+        solve_collisions();
+    }
+}
+
+void Scene::emplace_collider(std::shared_ptr<Collider2D> const& collider)
+{
+    std::weak_ptr<Collider2D> c_weak = collider;
+
+    // Check if the weak pointer is already in the vector
+    auto const it = std::find_if(colliders.begin(), colliders.end(),
+        [&](const std::weak_ptr<Collider2D>& wp) {
+            return !wp.expired() && wp.lock() == c_weak.lock();
+        });
```

Comment:
Since we are now adding / removing colliders in initialize / uninitialize this should not happen

These also don't need to be weak_ptrs anymore, if we want